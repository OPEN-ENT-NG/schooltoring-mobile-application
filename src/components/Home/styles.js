import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

const half = {
  flexDirection: "row",
  alignItems: "center",
  padding: 20
};

const image = {
  position: "absolute",
  bottom: 0
};

export default StyleSheet.create({
  topImageContainer: {
    ...image,
    right: 10
  },
  bottomImageContainer: {
    ...image,
    left: 10
  },
  text: {
    color: "white",
    fontSize: 24,
    width: "60%"
  },
  textLeft: {
    textAlign: "left"
  },
  textRight: {
    textAlign: "right"
  },
  topHalf: {
    ...half,
    backgroundColor: COLORS.SECONDARY,
    justifyContent: "flex-start"
  },
  bottomHalf: {
    ...half,
    backgroundColor: COLORS.PRIMARY,
    justifyContent: "flex-end"
  }
});
