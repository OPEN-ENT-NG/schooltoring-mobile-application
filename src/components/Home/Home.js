import React, { Component } from "react";
import { View, Text, Image, Dimensions } from "react-native";
import Proptypes from "prop-types";
import EventTracker from "../../api/EventTracker";

import Touchable from "../../components/Touchable/Touchable";
import styles from "./styles";

import I18n from "react-native-i18n";

const screen = Dimensions.get("screen");

export default class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      view: {
        width: screen.width,
        height: screen.height - 70
      }
    };
  }

  componentWillUnmount() {
    this._navListener.remove();
  }

  render() {
    return (
      <View
        style={{
          flex: 1,
          flexDirection: "column",
          alignItems: "center",
          justifyContent: "center"
        }}
        onLayout={event => {
          this.setState({
            view: {
              height: event.nativeEvent.layout.height,
              width: event.nativeEvent.layout.width
            }
          });
        }}
      >
        <Touchable
          onPress={() => {
            EventTracker.trackEvent(
              EventTracker.events.SEEK_HELP.click,
              EventTracker.category.SEEK_HELP
            );
            this.props.navigation.push("Strength");
          }}
        >
          <View
            style={[
              styles.topHalf,
              {
                width: this.state.view.width,
                height: this.state.view.height / 2
              }
            ]}
          >
            <Text style={[styles.text, styles.textLeft]}>
              {I18n.t("needHelp")}
            </Text>
            <View style={styles.topImageContainer}>
              <Image
                style={{ height: 250, width: 202 }}
                resizeMode="contain"
                source={require("../../assets/img/gars2b.png")}
              />
            </View>
          </View>
        </Touchable>
        <Touchable
          onPress={() => {
            EventTracker.trackEvent(
              EventTracker.events.OFFER_HELP.click,
              EventTracker.category.OFFER_HELP
            );
            this.props.navigation.push("Weakness");
          }}
        >
          <View
            style={[
              styles.bottomHalf,
              {
                width: this.state.view.width,
                height: this.state.view.height / 2
              }
            ]}
          >
            <View style={styles.bottomImageContainer}>
              <Image
                style={{ height: 250, width: 232 }}
                resizeMode="contain"
                source={require("../../assets/img/fille2.png")}
              />
            </View>
            <Text style={[styles.text, styles.textRight]}>
              {I18n.t("offerHelp")}
            </Text>
          </View>
        </Touchable>
      </View>
    );
  }
}

Home.proptypes = {
  navigation: Proptypes.any.isRequired
};
