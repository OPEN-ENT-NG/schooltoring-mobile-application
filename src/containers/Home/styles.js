import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

const half = {
  flex: 1,
  flexDirection: "row",
  alignItems: "center",
  padding: 20
};

const image = {
  position: "absolute",
  bottom: 0
};

export default StyleSheet.create({
  topImageContainer: {
    ...image,
    right: 10
  },
  bottomImageContainer: {
    ...image,
    left: 10
  },
  text: {
    color: "white",
    fontSize: 28
  },

  topHalf: {
    ...half,
    backgroundColor: COLORS.SECONDARY,
    justifyContent: "flex-start"
  },
  bottomHalf: {
    ...half,
    backgroundColor: COLORS.PRIMARY,
    justifyContent: "flex-end"
  }
});
