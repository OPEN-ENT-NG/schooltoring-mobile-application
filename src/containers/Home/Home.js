import React, { Component } from "react";
import { createStackNavigator } from "react-navigation";

import Home from "../../components/Home/Home";
import MatchList from "../MatchList/MatchList";
import Header from "../../components/Header/Header";

import I18n from "../../api/I18n";
import { COLORS } from "../../styles/common";

const getHeader = navigation => {
  let routeName = navigation.state.routeName;
  let noBack = navigation.getParam("noBack", false);
  switch (routeName) {
    case "Weakness": {
      return (
        <Header
          navigation={navigation}
          backgroundColor={COLORS.PRIMARY}
          noBack={noBack}
          title={I18n.t(`${routeName.toLowerCase()}.header`)}
        />
      );
    }
    case "Strength": {
      return (
        <Header
          navigation={navigation}
          backgroundColor={COLORS.SECONDARY}
          noBack={noBack}
          title={I18n.t(`${routeName.toLowerCase()}.header`)}
        />
      );
    }
    default: {
      return null;
    }
  }
};

const Stack = createStackNavigator(
  {
    Home: Home,
    Weakness: MatchList,
    Strength: MatchList
  },
  {
    initialRouteName: "Home",
    initialRouteParams: { noBack: true },
    cardStyle: { backgroundColor: COLORS.BACKGROUND },
    navigationOptions: ({ navigation }) => ({
      header: getHeader(navigation)
    })
  }
);

export default class HomeNavigator extends Component {
  render() {
    return <Stack />;
  }
}
