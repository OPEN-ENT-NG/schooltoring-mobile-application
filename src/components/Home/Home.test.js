import React from "react";
import { shallow } from "enzyme";

import Home from "./Home";

const navigation = {
  push: jest.fn().mockImplementation(routeName => routeName)
};

const rendered = shallow(<Home navigation={navigation} />);
describe("Render", () => {
  test("It should render without crashing", () =>
    expect(rendered).toMatchSnapshot());

  test("It should render 2 buttons", () => {
    expect(rendered.find("Touchable").length).toEqual(2);
  });
});

describe("Interactions", () => {
  test("On press on top button should trigger navigation.push function with 'Strength' parameter", () => {
    navigation.push.mockReset();

    const button = rendered.find("Touchable").at(0);
    button.props().onPress();
    expect(navigation.push).toHaveBeenCalled();
    expect(navigation.push.mock.calls[0][0]).toBe("Strength");
  });

  test("On press on bottom button should trigger navigation.push function with 'Weakness' parameter", () => {
    navigation.push.mockReset();

    const button = rendered.find("Touchable").at(1);
    button.props().onPress();
    expect(navigation.push).toHaveBeenCalled();
    expect(navigation.push.mock.calls[0][0]).toBe("Weakness");
  });
});
