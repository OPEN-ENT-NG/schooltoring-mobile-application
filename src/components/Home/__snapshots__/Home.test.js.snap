// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render It should render without crashing 1`] = `
<Component
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Touchable
    onPress={[Function]}
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#e91e63",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 20,
          },
          Object {
            "height": 632,
            "width": 750,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
              "width": "60%",
            },
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        [missing "en.needHelp" translation]
      </Text>
      <Component
        style={
          Object {
            "bottom": 0,
            "position": "absolute",
            "right": 10,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 250,
              "width": 202,
            }
          }
        />
      </Component>
    </Component>
  </Touchable>
  <Touchable
    onPress={[Function]}
  >
    <Component
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#17b6b0",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "padding": 20,
          },
          Object {
            "height": 632,
            "width": 750,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "bottom": 0,
            "left": 10,
            "position": "absolute",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 250,
              "width": 232,
            }
          }
        />
      </Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
              "width": "60%",
            },
            Object {
              "textAlign": "right",
            },
          ]
        }
      >
        [missing "en.offerHelp" translation]
      </Text>
    </Component>
  </Touchable>
</Component>
`;
